modules/product/src/main/java/com/opengamma/strata/product/swap/type/ImmutableFixedInflationSwapConvention.java

```diff
@@ -0,0 +1,639 @@
+  /**
+   * The offset of the spot value date from the trade date.
+   * <p>
+   * The offset is applied to the trade date to find the start date.
+   * A typical value is "plus 2 business days".
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final BusinessDayAdjustment spotDateOffset;
+  /**
+   * An adjustment that alters the payment date by adding a period of days.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final DaysAdjustment paymentDateOffset;
+
+  //-------------------------------------------------------------------------
+
+  /**
+   * Obtains a convention based on the specified name and leg conventions.
+   * <p>
+   * The two leg conventions must be in the same currency.
+   * 
+   * @param name  the unique name of the convention 
+   * @param fixedLeg  the market convention for the fixed leg
+   * @param floatingLeg  the market convention for the floating leg
+   * @param spotDateOffset  the offset of the spot value date from the trade date
+   * @param paymentDateOffset an adjustment that alters the payment date by adding a period of days
+   * @return the convention
+   */
+  public static ImmutableFixedInflationSwapConvention of(
+      String name,
+      FixedRateSwapLegConvention fixedLeg,
+      InflationRateSwapLegConvention floatingLeg,
+      BusinessDayAdjustment spotDateOffset,
+      DaysAdjustment paymentDateOffset) {
+
+    return new ImmutableFixedInflationSwapConvention(name, fixedLeg, floatingLeg, spotDateOffset, paymentDateOffset);
+  }
+
+  //-------------------------------------------------------------------------
+  @ImmutableValidator
+  private void validate() {
+    ArgChecker.isTrue(fixedLeg.getCurrency().equals(floatingLeg.getCurrency()), "Conventions must have same currency");
+  }
+
+  //-------------------------------------------------------------------------
+  @Override
+  public SwapTrade toTrade(
+      TradeInfo tradeInfo,
+      LocalDate startDate,
+      LocalDate endDate,
+      Period lag,
+      BuySell buySell,
+      double notional,
+      double fixedRate) {
+
+    Optional<LocalDate> tradeDate = tradeInfo.getTradeDate();
+    if (tradeDate.isPresent()) {
+      ArgChecker.inOrderOrEqual(tradeDate.get(), startDate, "tradeDate", "startDate");
+    }
+    SwapLeg leg1 = fixedLeg.toLeg(startDate, endDate, PayReceive.ofPay(buySell.isBuy()), notional, fixedRate);
+    SwapLeg leg2 = floatingLeg.toLeg(
+        startDate, 
+        endDate, 
+        PayReceive.ofPay(buySell.isSell()), 
+        lag, 
+        spotDateOffset, 
+        paymentDateOffset, 
+        notional);
+    return SwapTrade.builder()
+        .info(tradeInfo)
+        .product(Swap.of(leg1, leg2))
+        .build();
+  }
+
+  //-------------------------------------------------------------------------
+  @Override
+  public String toString() {
+    return getName();
+  }
+  //------------------------- AUTOGENERATED START -------------------------
```

Comment:
Should be a newline between the last method and the `----- AUTOGENERATED ---` line